packages/sitecore-jss-react/src/components/BYOCComponent.tsx

```diff
@@ -131,33 +131,31 @@ export class BYOCComponent extends React.Component<BYOCComponentProps> {
 
     const ErrorComponent = this.props.errorComponent;
 
-    const isNull = props.fetchedData && Object.keys(props.fetchedData).length === 0;
-
-    let componentProps: { [key: string]: any } = isNull ? null : props.fetchedData;
-
-    if (!componentProps) {
-      if (props.params?.ComponentProps) {
-        try {
-          componentProps = JSON.parse(props.params.ComponentProps) ?? {};
-        } catch (e) {
-          console.error(
-            `Parsing props for ${componentName} component from rendering params failed. Error: ${e}`
-          );
-          return ErrorComponent ? (
-            <ErrorComponent error={e as Error} />
-          ) : (
-            <DefaultErrorComponent error={e as Error} />
-          );
-        }
-      } else {
-        componentProps = props.fields ? getDataFromFields(props.fields) : {};
+    let componentProps: { [key: string]: any } = null;
+
+    if (props.params?.ComponentProps) {
+      try {
+        componentProps = JSON.parse(props.params.ComponentProps) ?? {};
+      } catch (e) {
+        console.error(
+          `Parsing props for ${componentName} component from rendering params failed. Error: ${e}`
+        );
+        return ErrorComponent ? (
+          <ErrorComponent error={e as Error} />
+        ) : (
+          <DefaultErrorComponent error={e as Error} />
+        );
       }
+    } else {
+      componentProps = props.fields ? getDataFromFields(props.fields) : {};
     }
+
     // we render fallback on client to avoid problems with client-only components
     return (
       <FEAAS.ExternalComponent
         componentName={componentName}
         clientFallback={fallbackComponent}
+        datasources={props.fetchedData}
         {...componentProps}
```

Comment:
might want to omit `datasources` from here since i imagine that shouldn't be overridden

Reply:
I think that `datasources` makes sense here, as componentProps is not the `props`, it's `props.params.ComponentProps`